acados/ocp_nlp/ocp_nlp_common.c

```diff
@@ -972,6 +978,13 @@ ocp_nlp_memory *ocp_nlp_memory_assign(ocp_nlp_config *config, ocp_nlp_dims *dims
     ocp_nlp_memory *mem = (ocp_nlp_memory *) c_ptr;
     c_ptr += sizeof(ocp_nlp_memory);
 
+    // set_sim_guess
+    assign_and_advance_bool(N+1, &mem->set_sim_guess, &c_ptr);
+    for (int ii = 0; ii <= N; ++ii)
+    {
+        mem->set_sim_guess[ii] = false;
+    }
+
```

Comment:
Always remember to align after you allocate an array of bool (like here, so I only want to warn). It is actually a bit tricky, because `sizeof(bool)` may be 1, 2 or 4 bytes depending on the compiler (even if just 1 bit is enough to hold the 0-1 information). In case of 1 byte size, if alignment doesn't happen after, it will screw up what is coming after and give rise to nasty bugs on certain architectures/compilers. So this may be a reason to prefer int which I didn't consider before.